### PR TITLE
Fix Windows IL2CPP compilation with dlopen/dlsym stubs

### PR DESCRIPTION
**Pull request by AcelisWeaven**
This pull request fixes and issue with IL2CPP and Windows, where GRPC needs the `dlopen` and `dlsym` functions. (see #2595)

Since we don't want to recompile GRPC from scratch or include it fully, the solution here is to make stubs for the missing functions. The .meta makes it so it's only compiled on Windows by IL2CPP.

This sounds like an ugly hack and may not be wanted. Thoughts?

_Help wanted:_ I don't really know how to properly name this file. Any suggestion?

### CODE REVIEW

**Comment by surfnerd** [comment]
Hey @AcelisWeaven,
Thanks for your contribution.  Does this break any other behavior on windows?  Does training still work in the editor with this?  Could you provide a little more information on how you tested this?  
Thanks!
Chris

**Comment by AcelisWeaven** [comment]
> Does training still work in the editor with this?

Yes it does. There should be no issue here, the file is marked as Standalone-only.
image

> Does this break any other behavior on windows?

Sadly there is an issue I forgot to mention : The compilation works, but when running as a standalone, it runs in Inference mode.
`Couldn't connect to trainer on port 5005 using API version API-14-dev0. Will perform inference instead.`
Switching back to Mono and it works. But I'm not sure the issue is related to this PR. Also, back on v0.8, I had a similar issue because of code stripping being enable (I don't know if this has been solved). With IL2CPP is can only set it to Low or higher. I've been trying IL2CPP Release and Debug.

> Could you provide a little more information on how you tested this?

I was able to compile both a custom environment and a freshly-cloned Tennis (both envs are similar), whereas I couldn't before. Both only work in Editor mode only with IL2CPP.

Here's the command I use for in-editor training:
```
```
Standalone training:
```
mlagents-learn.exe trainer_config.yaml --train --env=MyEnv --save-freq=10000 --run-id=myrun
```

TLDR: Inference and training in editor works, training with a standalone build doesn't :)

**Comment by surfnerd** [comment]
Hey @AcelisWeaven,
Thanks for the information!  Very helpful. I will bring this up with the team and consider merging it.

**Comment by surfnerd** [comment]
Hey @AcelisWeaven,
Can you update the comment to make it clear that training still doesn't work with this change?  After that, I will push a branch to our repo with your commit and merge it.

**Comment by AcelisWeaven** [comment]
@surfnerd Done! Thanks.

**Comment by caioc2** [comment]
Hi,

As someone who have tried out stubs for this problem, here are my 2 cents:

Stubs does not work when building the standalone to run multiple environments for training, which is the main point of ever using IL2CPP, **speed up training**.
Using it in the editor makes not much sense, since the Unity editor uses  Mono backend.

While the IL2CPP is a much needed fix, I think this is not the fix. It is a difficult issue which may be only solved with having directives in the GRPC and or some other clever idea to be found.

**Comment by surfnerd** [comment]
Hi @caioc2,
I agree that this doesn't help with the training aspect of this problem. Yet it would still be beneficial to anyone who is running inference and wants to build for IL2CPP.  You still get performance benefits by building with IL2CPP.  We fully understand that this doesn't fix the training issue, which is why we asked for the extra comment.  

That being said, we will be spending more time fixing the IL2CPP training issue.  Thanks for you comment.

**Comment by surfnerd** [comment]
Moved the PR here #3330
